pyacq/core/tools.py

```diff
@@ -22,10 +22,13 @@ class ThreadPollInput(QtCore.QThread):
     The `process_data()` method may be reimplemented to define other behaviors.
     """
     new_data = QtCore.Signal(int,object)
-    def __init__(self, input_stream, timeout=200, parent=None):
+    def __init__(self, input_stream, timeout=200, return_data=None, parent=None):
         QtCore.QThread.__init__(self, parent)
         self.input_stream = weakref.ref(input_stream)
         self.timeout = timeout
+        self.return_data = return_data
+        if self.return_data is None:
+            self.return_data = self.input_stream()._own_buffer
```

Comment:
So we get `return_data == False` for a plaindata stream that has no ring buffer?

Here's an idea to consider: what if `sharedmemstream.recv()` returns an object that looks just like an array, but defers concatenation unless the user actually tries to access the data? This would make it so that all `recv()` calls can have the same API and `return_data` would become obsolete. Something along the lines of:

    class MultipartArray:
        def __init__(self, parts):
            self._array_parts = parts
            self._array = None
            self.shape = (sum([p.shape[0] for p in parts],) + parts[0].shape[1:]
            self.dtype = parts[0].dtype

        def __getattr__(self, name):
            if self._array is None:
                self._array = np.concatenate(self._array_parts, axis=0)
            return getattr(self._array, name)